server/src/main/java/org/elasticsearch/script/field/DelegateDocValuesField.java

```diff
@@ -11,13 +11,13 @@
 import org.elasticsearch.index.fielddata.ScriptDocValues;
 
 import java.io.IOException;
-import java.util.List;
+import java.util.Iterator;
 
 /**
  * A default {@link Field} to provide {@code ScriptDocValues} for fields
  * that are not supported by the script fields api.
  */
-public class DelegateDocValuesField implements DocValuesField {
+public class DelegateDocValuesField implements DocValuesField<Object> {
```

Comment:
Is this going to work for `Numeric`s like long fields?

Reply:
Yes, ScriptDocValues type underneath is independent of the iterator type here which throws an UnsupportedOperationException.